phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -1565,6 +1589,13 @@ private List<Mutation> generateOnDupMutations(BatchMutateContext context, Put at
       // read the column values requested in the get from the current data row
       List<Cell> cells = IndexUtil.readColumnsFromRow(currentDataRowState, colsReadInExpr);
 
+      // store current cells into a map where the key is column family and the value is a map with
+      // column qualifier as key and a pair of cell and a boolean (is true if the expression is
+      // CaseExpression and Else-clause is evaluated to be true, is null if there is no
+      // expression on this column, otherwise false)
+      Map<ImmutableBytesPtr, Map<ImmutableBytesPtr, Pair<Cell, Boolean>>> currCfCqCellExprMap =
```

Comment:
We should use ColumnReference as the key here. It is already being used in this file. It could simplify the map here.

Reply:
Done